mantidproject/mantid
# Stop SliceViewer stacking delayed refreshes

**Description of work.**
SliceViewer currently utilises a delayed refresh. If a workspace with SliceViewer open is changed multiple times in quick succession these delayed refreshes stack and cause workbench to hang whilst they are processed.

This PR adds a simple flag to the delayed refresh to limit the amount of refreshes which can be queued.

**To test:**

1.  Run `ws = CreateSampleWorkspace()`
2.  Right click the created workspace and open sliceviewr.
3. Run:
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

ws = CreateSampleWorkspace()

for i in range(50):
    print(i)
    ws = ws*10
```
4. Observe minimal delay after execution of calculations.

*There is no associated issue.*

*This does not require release notes* because it fixes an issue caused by a recent commit.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
